nodejs/eks/examples/managed-nodegroups/iam.ts

```diff
@@ -0,0 +1,27 @@
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+
+const managedPolicyArns: string[] = [
+    "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+    "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
+    "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+];
+
+// Creates a role and attches the EKS worker node IAM managed policies
+export function createRole(name: string): aws.iam.Role {
+    const role = new aws.iam.Role(name, {
+        assumeRolePolicy: aws.iam.assumeRolePolicyForPrincipal({
+            Service: "ec2.amazonaws.com",
+        }),
+    });
+
+    let counter = 0;
```

Comment:
What is `counter` for?

Reply:
Looks like this creates a unique name for each RolePolicyAttachment that gets generated.

Reply:
Yes its for unique names. We've used this code before in other examples:

- https://github.com/pulumi/pulumi-eks/blob/master/nodejs/eks/examples/nodegroup/iam.ts
- https://github.com/pulumi/pulumi-eks/blob/master/nodejs/eks/examples/tests/migrate-nodegroups/iam.ts